### PR TITLE
Starter blog

### LINKED ISSUE #1615: Blog about the Open Liberty Starter
Let's write a blog on yet another way to get started on Open Liberty. We should discuss the starter in the first half and then summarize other ways someone can get started with Open Liberty like using a docker image, maven/gradle guides, downloading the binary, etc.

### PR DESCRIPTION
**Pull request by steven1046**
https://blogs-draft-openlibertyio.mybluemix.net/blog/2021/08/31/open-liberty-starter.html

Closes #1615

### CODE REVIEW

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (mbroz2) [inline posts/2021-08-31-open-liberty-starter.adoc]: set this to your image in the blog

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (mbroz2) [inline posts/2021-08-31-open-liberty-starter.adoc]: I think we should move this image up to right after your first paragraph (aka, after the 'then just click generate' part.

**Review by mbroz2 — CHANGES_REQUESTED** [review]

**Comment by steven1046** [comment]
The requested changes are now on the blog draft site.

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (mbroz2) [inline posts/2021-08-31-open-liberty-starter.adoc]: I'm not sure we need the "Maven and Gradle" section.  I think just drop it and make the individual "Maven" and "Gradle" sections peers of the other ones (===)

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (mbroz2) [inline posts/2021-08-31-open-liberty-starter.adoc]: Wondering if we want "Binary" or "Package" or "Binary Package".... we do call it a "Package" on /downloads so go with one of the latter two.

**Review by mbroz2 — CHANGES_REQUESTED** [review]

**Comment by mbroz2** [comment]
@dmuelle can you review as well when you get a chance.

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (dmuelle) [inline posts/2021-08-31-open-liberty-starter.adoc]: `Visit the link:{url-prefix}/downloads/[Getting Started] page and select a few options to generate a starter application using Open Liberty.`

minor nit- this is a little ambiguous and might be read as OL itself generating the app, Suggested edit:

`Visit the link:{url-prefix}/downloads/[Getting Started] page and select a few options to generate a starter application that runs on Open Liberty.`

or similar....

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (dmuelle) [inline posts/2021-08-31-open-liberty-starter.adoc]: image needs alt text, something like:

`image::img/blog/starter_application.png[screen capture of the starter application options,align="center"]`

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (dmuelle) [inline posts/2021-08-31-open-liberty-starter.adoc]: is it overkill to say something like "...open a command line session, navigate to the installation directory, and run the `mvn liberty:run` command"

yes most users will know this is a command line instruction but since this is a "getting started" we might want to spell it out for new users

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (dmuelle) [inline posts/2021-08-31-open-liberty-starter.adoc]: one question that is not addressed here: what is a starter app? If someone is not already familiar with the concept, they won't be able to figure it out from this page.

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (dmuelle) [inline posts/2021-08-31-open-liberty-starter.adoc:51]: Should there be a few lines here to let the user know what to expect, or what the starter app can do? What sort of confirmation/feedback would the user expect to see if they are successful? How do they know it worked? What should they do next?

The links that follow don't directly address working with a starter app, but are more general getting started information. I think that distinction should be more clear.

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (dmuelle) [inline posts/2021-08-31-open-liberty-starter.adoc]: There is a guide on using Docker with Open Liberty link:{url-prefix}/guides/containerize.html[here].

---->

For more information, see our guide on  using Docker with Open Liberty, link:{url-prefix}/guides/containerize.html[Containerizing microservices]

or would [Using Docker containers to develop microservices](https://www.openliberty.io/guides/docker.html) be a better reference?

**Review by mbroz2 — DISMISSED** [review]
Approved, but wait for feedback from @dmuelle before delivering.

**Comment by steven1046** [comment]
> Approved, but wait for feedback from @dmuelle before delivering.

Ok, i was waiting to fix all feedback into draft before merging here.

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (dmuelle) [inline posts/2021-08-31-open-liberty-starter.adoc]: For these commands- is that a zsh syntax to include `./` in the command? This might be a larger style question but we haven't done that elsewhere in the docs. And would the Maven  command in the first section also use that syntax?

**Review by dmuelle — COMMENTED** [review]
Looks good overall. I wonder if we need to explain the concept of a starter app (in brief). 

Also there are several different senses of "Getting started" in this post that are not necessarily the same thing. Or maybe they are ?

Is using a starter app the same as building Liberty with Maven/Gradle/Docker? It might be more clear to focus more on the starter app, why you'd want it and what you can do with it, then just provide links to the other guides at the end  as a reference.

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (dmuelle) [inline posts/2021-08-31-open-liberty-starter.adoc]: I think this description looks odd as the preview line in the blog roll- doesn't tell the reader much

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (mbroz2) [reply posts/2021-08-31-open-liberty-starter.adoc]: I think you're right that we should remove the './' as that is OS dependent.  Perhaps instead we can just say:
Then navigate to the `bin` directory and execute the `server create <server_name>` command to create an Open Liberty server, or `server start` command to automatically create and start a `defaultServer`.

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (mbroz2) [reply posts/2021-08-31-open-liberty-starter.adoc:51]: good point, we should at least tell them what they'll see when the process has finished.

Comment (mbroz2) [reply posts/2021-08-31-open-liberty-starter.adoc:51]: I think that could include both the final output of the command line (smarter planet msg), as well as a view of the directory structure in an IDE

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (mbroz2) [reply posts/2021-08-31-open-liberty-starter.adoc]: agreed, we should expand on what the starter is/provides and why someone would want to use it.

Comment (steven1046) [reply posts/2021-08-31-open-liberty-starter.adoc]: Good point!

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (dmuelle) [inline posts/2021-08-31-open-liberty-starter.adoc]: A simple RestApplication.java gets generated for you to start creating a project. A server.xml configuration file gets generated for you with the necessary features for the MicroProfile and Jakarta EE versions that you pick.
---->
A simple `RestApplication.java` file iss generated for you to start creating a project. A `server.xml`  configuration file is also generated  with the necessary features for the MicroProfile and Jakarta EE versions that you selected during set up.

**Review by dmuelle — COMMENTED** [review]
minor edit suggestions

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (dmuelle) [inline posts/2021-08-31-open-liberty-starter.adoc]: per Michals repsonse to my earlier comment- should we remove `./` from command example syntax, since it's OS dependent? Here and in the previous Gradle command example

Comment (steven1046) [reply posts/2021-08-31-open-liberty-starter.adoc]: Right, sorry i thought i fixed it! I will update this in the next commit.

File: posts/2021-08-31-open-liberty-starter.adoc
Comment (mbroz2) [reply posts/2021-08-31-open-liberty-starter.adoc]: A simple RestApplication.java file is generated for you to start creating a REST based application. A server.xml configuration file is also provided with the necessary features for the MicroProfile and Jakarta EE versions that you previously selected.
If you plan on developing and/or deploying your app in a containerized environment, the included Dockerfile will make it easier to create your application image on top of the Open Liberty Docker image.

**Review by mbroz2 — CHANGES_REQUESTED** [review]

**Comment by steven1046** [comment]
https://blogs-draft-openlibertyio.mybluemix.net/blog/2021/08/31/open-liberty-starter.html

**Review by dmuelle — APPROVED** [review]
LGTM

**Comment by steven1046** [comment]
According to the instructions in the README, that's all i have to do right? The date can be changed to be published earlier.

**Comment by dmuelle** [comment]
yes- except the commits also need to go back to draft to keep things in sync. And the tags.json file needs to be updated

**Comment by steven1046** [comment]
All of the commits are in draft too. Will the editors handle the tags?

**Comment by dmuelle** [comment]
I can update them.

**Comment by mbroz2** [comment]
For the tags, I suggest: "developer experience", "Jakarta EE", "MicroProfile", "Gradle", "Maven", and "Docker"

**Comment by dmuelle** [comment]
https://github.com/OpenLiberty/blogs/pull/1639

**Review by mbroz2 — APPROVED** [review]

**Pull request merged**
